### PR TITLE
7 more coverity bugs fixed

### PR DESCRIPTION
**Pull request by wargio**
* 1379624 Uninitialized scalar variable
* 1379743 Use after free
* 1379751 Double free
* 1379752 Resource leak
* 1379742/1379747/1379748/1379749/1379750/1379747/1379760 Resource leak
* 1379834 Resource leak
* 1379842/1379836 Resource leak

### CODE REVIEW

**Comment by XVilka** [comment]
I suggest we wait until master is green. Fixing coverities sometimes can lead to segfaults.

File: libr/bin/bin.c [hunk 1]
```diff
@@ -1092,10 +1092,12 @@ R_API int r_bin_load_io_at_offset_as_sz(RBin *bin, int fd, ut64 baseaddr,
 					} else if (sz != file_sz) {
 						iob->read_at (io, 0LL, buf_bytes, sz);
 					}
-					binfile = r_bin_file_xtr_load_bytes (bin, xtr,
-						fname, buf_bytes, sz, file_sz,
-						baseaddr, loadaddr, xtr_idx,
-						fd, bin->rawstr);
+					if (buf_bytes) {
```
Comment (radare) [inline libr/bin/bin.c]: imho r_bin_file_xtr_load_bytes should check for that null , not the caller

Comment (wargio) [reply libr/bin/bin.c]: I thought the same but I wasn't sure it was used somewhere else. I'll patch it

Comment (radare) [reply libr/bin/bin.c]: ok ill merge after that. r2 will shine soon in travisland

**Comment by radare** [comment]
tired of waiting

File: libr/bin/bin.c (hunk 1 above)
Comment (radare) [reply libr/bin/bin.c]: fixed that in my commit after you

**Comment by wargio** [comment]
sorry. i've been busy. i'll make a new PR for this.

**Pull request merged**